### PR TITLE
[Netatmo] Do not refresh module data when not expected

### LINKED ISSUE #3286: Netatmo: call of the netatmo API each time a Welcome Home channel is linked
Each time a channel is linked (using Paper UI), the binding is calling the API with a `getlasteventof` request.
It might be the reason why I was banished few days ago.
I think the binding should cache the last API call result and use it when linking new channels.

@clinique: WDYT ?

### PR DESCRIPTION
**Pull request by lolodomo**
Fixes #3286

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@clinique for a review.

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
     private ScheduledFuture<?> refreshJob;
     @Nullable
     protected MODULE module;
+    protected boolean refreshRequired;
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java]: Can it be private?

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java [hunk 2]
```diff
@@ -97,8 +98,17 @@ protected void updateChannels(Object module) {
     }
 
     protected void requestParentRefresh() {
+        setRefreshRequired(true);
         Optional<AbstractNetatmoThingHandler> parent = getBridgeHandler().findNAThing(getParentId());
         parent.ifPresent(AbstractNetatmoThingHandler::updateChannels);
     }
 
+    public boolean isRefreshRequired() {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java]: Would protected access be enough?

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java [hunk 3]
```diff
@@ -97,8 +98,17 @@ protected void updateChannels(Object module) {
...
+        return refreshRequired;
+    }
+
+    public void setRefreshRequired(boolean refreshRequired) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java]: See above

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java (hunk 1 above)
Comment (lolodomo) [reply addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java]: Yes, right.

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java (hunk 2 above)
Comment (lolodomo) [reply addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoModuleHandler.java]: Yes, probably.

**Comment by lolodomo** [comment]
PR rebased to solve the conflict.

**Comment by lolodomo** [comment]
@clinique : have you the time to review this little but important change ?

**Comment by martinvw** [comment]
Thanks!

**Pull request merged**